lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -82,6 +84,34 @@ def __init__(self, charm: CharmBase, substrate="k8s", relation_name: str = "data
             self.database_provides.on.database_requested, self._on_relation_event
         )
 
+    def pass_hook_checks(self) -> bool:
+        """Runs the pre-hooks checks for MongoDBProvider, returns True if all pass."""
+        if self.charm.is_role(Config.Role.SHARD) or self.charm.is_role(Config.Role.CONFIG_SERVER):
+            self.charm.unit.status = BlockedStatus(
```

Comment:
imo, this method should not apply statuses, but rather just stick to returning booleans, and when calling it - depending on the return value to call a method with the exhaustive list of conditions + status. 
i.e: 
```
if not self.pass_hook_checks():
    logger.info("Skipping %s: hook checks did not pass", type(event))
    self.apply_status_if_needed()
    return

self.clear_status()  # this only targets relation related statuses
```

Reply:
I am happy to do that, but i think it would end up having the function copy pasted, just with/without status. I personally don't like this, but if you feel strongly about that implementation/style I will gladly do it :)